packages/flame_tiled/lib/src/renderable_tile_map.dart

```diff
@@ -192,17 +266,165 @@ class RenderableTiledMap {
     });
   }
 
-  /// Render [batchesByLayer] that compose this tile map.
-  void render(Canvas c) {
-    batchesByLayer.forEach((batchMap) {
-      batchMap.forEach((_, batch) => batch.render(c));
+  @override
+  void render(Canvas canvas) {
+    canvas.save();
+
+    if (_camera != null) {
+      _applyParallaxOffset(canvas, _camera!, layer);
+    }
+
+    _cachedSpriteBatches.values.forEach((batch) {
+      batch.render(canvas, paint: _layerPaint);
     });
+
+    canvas.restore();
   }
 
-  /// Returns a layer of type [T] with given [name] from all the layers
-  /// of this map. If no such layer is found, null is returned.
-  T? getLayer<T extends Layer>(String name) {
-    return map.layers
-        .firstWhereOrNull((layer) => layer is T && layer.name == name) as T?;
+  static Future<_RenderableLayer> load(
+    TileLayer layer,
+    TiledMap map,
+    Camera? camera,
+    Vector2 destTileSize,
+  ) async {
+    return _RenderableTileLayer(
+      layer,
+      map,
+      camera,
+      destTileSize,
+      await _loadImages(map),
+    );
+  }
+
+  static Future<Map<String, SpriteBatch>> _loadImages(TiledMap map) async {
+    final result = <String, SpriteBatch>{};
+
+    await Future.forEach(map.tiledImages(), (TiledImage img) async {
+      final src = img.source;
+      if (src != null) {
+        result[src] = await SpriteBatch.load(src);
+      }
+    });
+
+    return result;
+  }
+}
+
+class _RenderableImageLayer extends _RenderableLayer<ImageLayer> {
+  final Image _image;
+  late final ImageRepeat _repeat;
+  late final Camera? _camera;
+  Rect _paintArea = Rect.zero;
+
+  _RenderableImageLayer(super.layer, this._camera, this._image) {
+    _initImageRepeat();
+  }
+
+  @override
+  void handleResize(Vector2 canvasSize) {
+    _paintArea =
+        Rect.fromLTWH(0, 0, _camera!.canvasSize.x, _camera!.canvasSize.y);
+  }
+
+  @override
+  void render(Canvas canvas) {
+    canvas.save();
+
+    // this seems to match how the Tiled editor initially offsets image layers
+    canvas.translate(-_image.width + layer.offsetX, layer.offsetY);
+
+    _applyParallaxOffset(canvas, _camera!, layer);
+
+    paintImage(
+      canvas: canvas,
+      rect: _paintArea,
+      image: _image,
+      opacity: layer.opacity,
+      alignment: Alignment.topLeft,
+      repeat: _repeat,
+      filterQuality: FilterQuality.medium,
```

Comment:
Maybe this should be configurable?

Reply:
I'm going to remove this line of code so it just uses the default value (`FilterQuality.low`), I think making this configurable would be a little to complicated for now since it's not clear if anyone needs it